examples/demo_threading/src/lib.rs

```diff
@@ -0,0 +1,335 @@
+const SENSOR_TIMEOUT_MILLIS: u128 = 10_000;
+/// How often should the timeout thread poll sensors
+const SENSOR_TIMEOUT_POLL_RATE_MILLIS: u64 = 256;
+/// The maximum number of sensors we will manage
+const SENSOR_MAXIMUM_COUNT: usize = 1000;
+/// The maximum power a sensor can report
+const SENSOR_MAXIMUM_POWER: f64 = 1000.0;
+
+// Network Serialisation definition
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum RequestCommand {
+    Disconnect,
+    Power { value: f64 },
+}
+
+#[derive(Deserialize, Serialize)]
+struct Request {
+    command: RequestCommand,
+    uuid: Uuid,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Status {
+    Ok,
+    ErrorInvalidPower,
+    ErrorInvalidRequest,
+    ErrorServerQueueFull,
+    ErrorServerDisconnected,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Response {
+    status: Status,
+}
+
+impl From<Status> for Response {
+    fn from(status: Status) -> Self {
+        Response { status }
+    }
+}
+
+impl From<Result<(), std::sync::mpsc::TrySendError<NetworkChannel>>> for Response {
+    fn from(result: Result<(), std::sync::mpsc::TrySendError<NetworkChannel>>) -> Self {
+        match result {
+            Ok(_) => Response { status: Status::Ok },
+            Err(TrySendError::Full { .. }) => Response {
+                status: Status::ErrorServerQueueFull,
+            },
+            Err(TrySendError::Disconnected { .. }) => Response {
+                status: Status::ErrorServerDisconnected,
+            },
+        }
+    }
+}
+
+// Channel definition
+enum NetworkChannel {
+    Disconnect { uuid: Uuid },
+    Power { uuid: Uuid, value: f64 },
+    TimeoutUpdate,
+}
+
+#[derive(Clone)]
+struct SensorData {
+    power: f64,
+    last_seen: SystemTime,
+}
+
+impl Default for SensorData {
+    fn default() -> Self {
+        Self {
+            power: 0.0,
+            last_seen: SystemTime::now(),
+        }
+    }
+}
+
+#[make_qobject]
+mod energy_usage {
+    use super::{NetworkChannel, Request, RequestCommand, Response, SensorData, Status};
+    use async_std::{
+        net::{TcpListener, TcpStream},
+        prelude::*,
+        task::spawn,
+    };
+    use futures::{executor::block_on, stream::StreamExt};
+    use futures_timer::Delay;
+    use std::{
+        collections::HashMap,
+        sync::mpsc::{sync_channel, Receiver, SyncSender},
+        thread::JoinHandle,
+        time::{Duration, SystemTime},
+    };
+    use uuid::Uuid;
+
+    pub struct Data {
+        average_use: f64,
+        sensors: u32,
+        total_use: f64,
+    }
+
+    impl Default for Data {
+        fn default() -> Self {
+            Self {
+                average_use: 0.0,
+                sensors: 0,
+                total_use: 0.0,
+            }
+        }
+    }
+
+    struct RustObj {
+        qt_rx: Receiver<Data>,
+        qt_tx: SyncSender<Data>,
+        join_handle_network: Option<JoinHandle<()>>,
```

Comment:
These three fields can probably be moved into a single:
`join_handles: Option<[JoinHandle<()>;3]>` member

Reply:
Done